accepted/future-releases/3009-inference-using-bounds/feature-specification.md

```diff
@@ -0,0 +1,319 @@
+  space:3
+  csstvHorizontalN["CSSTV"]
+  space:2
+  csstvUpwards["CSSTV"]
+  space
+  outputOverall["Output"]
+
+  space:15
+
+  space:2
+  boundsDownwards["Bounds*"]
+  space:2
+  boundsHorizontal1["Bounds*"]
+  space:3
+  boundsHorizontalN["Bounds*"]
+  space:2
+  boundsUpwards["Bounds*"]
+  space:2
+
+  downwards -- "Constraints" --> csstvDownwards
+  horizontal1 -- "Constraints" --> csstvHorizontal1
+  horizontalN -- "Constraints" --> csstvHorizontalN
+  upwards -- "Constraints" --> csstvUpwards
+
+  inputOverall --> csstvDownwards
+  csstvDownwards --> csstvHorizontal1
+  csstvHorizontal1 -- "..." --> csstvHorizontalN
+  csstvHorizontalN --> csstvUpwards
+  csstvUpwards --> outputOverall
+
+  boundsDownwards --> csstvDownwards
+  boundsHorizontal1 --> csstvHorizontal1
+  boundsHorizontalN --> csstvHorizontalN
+  boundsUpwards --> csstvUpwards
+```
+
+An important property of CSSTV is the following: if an element of the preliminary solution from the previous phase is known (that is, it doesn't contain `_`), it is not changed by the current run of CSSTV. Effectively, once a known type is produced at the end of any phase, it becomes "frozen" through the rest of the inference process and becomes a part of the overall inference result.
+
+The picture below shows the details of the type inference process for the motivating example using the schematic representation.
+
+```mermaid
+block-beta
+  columns 8
+  space downwards["Downwards phase"]:3 upwards["Upwards phase"]:3 space
+
+  space:4
+  newConstraint["X <: A<_>"]
+  space:3
+
+  overallInput["Input"]
+  space
+  csstvDownwards["CSSTV"]
+  space:2
+  csstvUpwards["CSSTV"]
+  space
+  overallOutput["Output"]
+
+  space:8
+
+  space:4
+  constraintGeneration["X <: A<_>"]
+  space:3
+
+  space:2
+  boundsDownwards["X extends A<X>"]
+  space:2
+  boundsUpwards["X extends A <X>"]
+  space:2
+
+  downwards -- "∅" --> csstvDownwards
+  upwards -- "C <: X" --> csstvUpwards
+
+  overallInput -- "{X = _}" --> csstvDownwards
+  csstvDownwards -- "{X = _}" --> csstvUpwards
+  csstvUpwards -- "{X = C}" --> overallOutput
+
+  constraintGeneration --> newConstraint
+  boundsUpwards --> constraintGeneration
+
+  boundsDownwards --> csstvDownwards
+  boundsUpwards --> csstvUpwards
+```
+
+## Proposed changes
+
+We propose generating more constraints from existing constraints and the bound. The original example is solved by this proposal and a few more beneficial examples are discussed afterwards.
+
+In the motivating example, adding the best-effort bound approximation didn't affect the result since the set of constraints for `X` already contained a lower-bound constraint for `X` (`C <# X`), and that constraint had the priority over the upper-bound "best-effort" approximation constraint, since type `C` is fully known (i.e. doesn't contain `_`). In the current proposal ***we aim to derive a lower-bound constraint by combining the already existing lower-bound constraint with the actual bound***.
```

Comment:
These sentences could be made a bit shorter, which would probably help the reader:
```suggestion
In the motivating example, adding the best-effort bound approximation didn't affect the result. The set of constraints for `X` already contained a lower-bound constraint for `X` (`C <# X`), and that constraint has priority over the upper-bound "best-effort" approximation constraint, since type `C` is fully known (i.e. doesn't contain `_`).

With this feature, we derive a lower-bound constraint by combining the already existing lower-bound constraint with the actual bound.
```

I split the paragraph in two because this allows the last sentence to be more prominent, without the unusual ***doubly embellished font*** usage.

Reply:
Thanks for the suggestion, Erik! I've updated the text accordingly.